src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -559,32 +568,43 @@ private List<File> scanPath(Path path, int pathIndex) {
      * @return the list of files included via the filesets.
      */
     protected List<File> scanFileSets() {
-        final List<File> allFiles = new ArrayList<>();
-
-        for (int i = 0; i < fileSets.size(); i++) {
-            final FileSet fileSet = fileSets.get(i);
-            final DirectoryScanner scanner = fileSet.getDirectoryScanner(getProject());
-            final List<File> scannedFiles = retrieveAllScannedFiles(scanner, i);
-            allFiles.addAll(scannedFiles);
-        }
-
-        return allFiles;
+        return fileSets
+            .stream()
+            .map(fileSet -> fileSet.getDirectoryScanner(getProject()))
+            .peek(this::logAdditionOfFilesFromDir)
+            .map(this::mapToAbsolutePaths)
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
     }
 
     /**
-     * Retrieves all matched files from the given scanner.
+     * Logs information about files being added from a directory scan.
      *
-     * @param scanner  A directory scanner. Note, that {@link DirectoryScanner#scan()}
-     *                 must be called before calling this method.
-     * @param logIndex A log entry index. Used only for log messages.
-     * @return A list of files, retrieved from the given scanner.
+     * @param scanner The file scanner containing the file information
      */
-    private List<File> retrieveAllScannedFiles(DirectoryScanner scanner, int logIndex) {
-        final String[] fileNames = scanner.getIncludedFiles();
-        log(String.format(Locale.ROOT, "%d) Adding %d files from directory %s",
-            logIndex, fileNames.length, scanner.getBasedir()), Project.MSG_VERBOSE);
+    private void logAdditionOfFilesFromDir(FileScanner scanner) {
+        log(String.format(
+                Locale.ROOT,
+                "Adding %d files from directory %s",
```

Comment:
```suggestion
                "%d) Adding %d files from directory %s",
```
Is the pre: number required? Because it seems not local tests found this, as my build was stable.